src/main/java/de/qabel/desktop/daemon/sync/worker/index/SyncState.java

```diff
@@ -0,0 +1,71 @@
+package de.qabel.desktop.daemon.sync.worker.index;
+
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+public class SyncState {
+    private boolean existing;
+    private Long mtime;
+    private Long size;
+
+    /**
+     * Creates default sync state (not synced at all = not existing, without information)
+     */
+    public SyncState() {
+        this(false, null, null);
+    }
+
+    public SyncState(boolean existing, Long mtime, Long size) {
+        this.existing = existing;
+        this.mtime = mtime;
+        this.size = size;
+    }
+
+    public boolean isExisting() {
+        return existing;
+    }
+
+    public Long getMtime() {
+        return mtime;
+    }
+
+    public Long getSize() {
+        return size;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof SyncState)) {
+            return false;
+        }
+
+        SyncState otherState = (SyncState) obj;
+        if (otherState.getMtime() == null && getMtime() != null && getMtime() != 0L) {
+            return false;
+        }
+        if (otherState.getSize() == null && getSize() != null && getSize() != 0L) {
+            return false;
+        }
+
+        return otherState.isExisting() == existing
+            && (otherState.getMtime() == null && getMtime() == null
+                || otherState.getMtime() == null && getMtime() == 0L
+                || otherState.getMtime() == 0L && getMtime() == null
+                || otherState.getMtime().equals(mtime))
```

Comment:
what about `nullOrZero(otherState.getMtime() == nullOrZero(getMtime())`?

I really don't like that Mtime can be a null pointer, same for size.